rawfile/utils/volume_manager.py

```diff
@@ -0,0 +1,229 @@
+from enum import Enum
+import time
+from typing import TypedDict
+import consts
+from utils.devices import device_to_mountpoint, path_stats
+from utils.errors import SourceTypeRequired
+from utils.lock import VolLock
+from utils.rawfile import (
+    attached_loops,
+    fallocate,
+    img_dir,
+    img_file,
+    meta_file,
+    metadata,
+    metadata_or,
+    patch_metadata,
+    snapshots_dir,
+    truncate,
+    update_metadata,
+)
+from pathlib import Path
+from utils.snapshot_manager import manager as snapshot_manager
+from os.path import basename, dirname, getsize
+from utils.logs import logger
+from volume_schema import LATEST_SCHEMA_VERSION, migrate_to
+from glob import glob
+
+
+class VolumeSource(int, Enum):
+    snapshot = 1
+    volume = 2
+
+
+class VolumeStats(TypedDict):
+    total: int
+    used: int
+
+
+class VolumeManager:
+    def _get_volume_path(self, volume_id: str) -> Path:
+        return img_dir(volume_id)
+
+    def create_volume(
+        self,
+        volume_id: str,
+        size: int,
+        thin_provision: bool,
+        freezefs: bool | None = None,
+        copy_on_write: bool | None = None,
+        source_type: VolumeSource | None = None,
+        source_id: str | None = None,
+    ):
+        if source_id and (not source_type):
+            raise SourceTypeRequired(source_id)
+        snapshot_name = None
+        source_volume_id = None
+        img_data_dir = img_dir(volume_id)
+        img_data_dir.mkdir(mode=consts.D_PERMS, exist_ok=True)
+        patch_metadata(volume_id, {"ready": False})
```

Comment:
Do we need this for parentless volumes?

Reply:
Since the entire creation is in the background, I have added this for all of the volumes